packages/ai-jsx/src/core/conversation.tsx

```diff
@@ -238,6 +238,45 @@ function toConversationMessages(partialRendering: AI.PartiallyRendered[]): Conve
   });
 }
 
+async function loggableMessage(
+  message: ConversationMessage,
+  render: AI.RenderContext['render'],
+  measureCost?: (message: ConversationMessage, render: AI.ComponentContext['render']) => Promise<number>
+) {
+  let textPromise: PromiseLike<string> | undefined = undefined;
+  switch (message.type) {
+    case 'user':
+    case 'assistant':
+    case 'system':
+      textPromise = render(message.element);
+      break;
+    case 'functionResponse':
+      textPromise = render(message.element.props.children);
+      break;
+    case 'functionCall':
+      break;
+    default: {
+      const neverMessage: never = message;
+      throw new Error(`Unexpected message type ${(neverMessage as any).type}`);
+    }
+  }
+
+  const costPromise = measureCost?.(message, render);
+  const loggableProps = { ...message.element.props };
+  if ('children' in loggableProps) {
+    delete loggableProps.children;
+  }
```

Comment:
Are all the other `props` Jsonifiable and `children` is the only one that gets rendered?

Reply:
Yup exactly. Will improve this to make it explicitly type-checked :)